nilearn/plotting/tests/test_html_stat_map.py

```diff
@@ -51,5 +42,89 @@ def test_view_stat_map():
         html = html_stat_map.view_stat_map(img_4d, threshold=2., vmax=4.)
         _check_html(html)
     warning_categories = set(warning_.category for warning_ in w)
-    expected_categories = set([FutureWarning, UserWarning])
-    assert_equal(warning_categories, expected_categories)
+    expected_categories = set([FutureWarning, UserWarning,
+                               DeprecationWarning])
+    _assert_warnings_in(warning_categories, expected_categories)
+
+
+def test_data2sprite():
+
+    # Generate a simulated volume with a square inside
+    data = np.zeros([8, 8, 8])
+    data[2:6, 2:6, 2:6] = 1
+
+    # turn that into a sprite and check it has the right shape
+    sprite = html_stat_map._data2sprite(data)
+    assert sprite.shape == (24, 24)
+
+    # Generate ground truth for the sprite
+    Z = np.zeros([8, 8])
+    Zr = np.zeros([2, 8])
+    Cr = np.tile(np.array([[0, 0, 1, 1, 1, 1, 0, 0]]), [4, 1])
+    C = np.concatenate((Zr, Cr, Zr), axis=0)
+    gtruth = np.concatenate((np.concatenate((Z, Z, C), axis=1),
+                             np.concatenate((C, C, C), axis=1),
+                             np.concatenate((Z, Z, Z), axis=1)),
+                            axis=0)
+
+    # Check that the sprite matches ground truth
+    assert (sprite == gtruth).all()
+
+
+def test_get_vmin_vmax():
+
+    # Generate a simulated volume with a square inside
+    data = np.zeros([8, 8, 8])
+    data[2:6, 2:6, 2:6] = 1
+
+    # Check default vmin, vmax
+    vmin, vmax = html_stat_map._get_vmin_vmax(data)
+    assert (vmin == 0) and (vmax == 1)
+
+    # Force vmin and vmax
+    vmin, vmax = html_stat_map._get_vmin_vmax(data, vmin=.5, vmax=.7)
+    assert (vmin == .5) and (vmax == .7)
+
+    # Try to use NaN for vmax
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
```
    assert_warns(
        UserWarning, html_stat_map._get_vmin_vmax, data, vmin=.5, vmax=np.nan)
    assert_warns(
        UserWarning, html_stat_map._get_vmin_vmax, data, vmin=np.nan, vmax=.7)
    assert_raises(
        ValueError, html_stat_map._get_vmin_vmax, data, vmin=3, vmax=0.7)
```

where `assert_warns` and `assert_raises` are imported from `numpy.testing`

Reply:
> There is still a test failing in python 2 btw, I'll look into that.

this should fix it

Reply:
> where assert_warns and assert_raises are imported from numpy.testing

Not from nosetests.tools?

Reply:
no because we might switch from nosetests to pytest at some point

Reply:
Discussion: Is it a great practice to use testing functions from non-test-runner? Pytest will run nose, and when making the switch I will make the change since I will be looking for functions from nose (but not numpy.testing). Then again, it may not matter. What do you think?

Reply:
Maybe we can write new tests using pytest, period. (?)

Reply:
sure. the point was more that here in python2 the same warning doesn't get raised twice, so this test is failing. moreover, the last part of the test isn't checking that an exception is actually being raised, just ignoring it if it is. any solution which relies on testing utils made for checking warnings and exceptions is fine by me.

Reply:
thanks for the suggestion @jeromedockes . Adding it.